### ISSUE TITLE
Passing context to functional components

### ISSUE THREAD

**Issue by eddyerburgh** [vuejs/vue-test-utils#12]
I propose this API to pass context to functional components:

```js
const context = {
  props: {foo: 'bar'}
}

const wrapper = mount(FunctionalComponent, context)
```

What are peoples thoughts?

**Comment by codebryo** [comment]
Agree to 100%, feels natural and allows for customization or extending in the future.
Would you internally turn it to `propsData`?

**Comment by eddyerburgh** [comment]
The way I got this working in avoriaz is to mount a component with a render functional that returns the functional component and passes it the context object -

https://github.com/eddyerburgh/avoriaz/blob/master/src/mount.js#L33

I imagine there are problems with doing it like this, but I wanted to get something out there as a canary test.

Maybe we should also allow users to pass props to `mount` with `props` instead of `propsData`, as it's cleaner

**Comment by codebryo** [comment]
I think it makes sense to make `props` work for both contexts. And it's a nicer API then using `propsData`.
Ideally it makes no difference what kind of Component gets mounted as all params will be passed in the same way. Options passed along that don't count for functional components will then be ignored by Vue instance anyway.

**Comment by eddyerburgh** [comment]
Closed with eda39df46c5e18d16aa8c16f9e29388640e31d96
